File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Prompts/PromptResponse.cs
using Microsoft.Teams.AI.AI.Models;
using Microsoft.Teams.AI.Exceptions;

namespace Microsoft.Teams.AI.AI.Prompts
{
    /// <summary>
    /// Response returned by a `PromptCompletionClient`.
    /// </summary>
    public class PromptResponse
    {
        /// <summary>
        /// Status of the prompt response.
        /// </summary>
        public PromptResponseStatus Status { get; set; }

        /// <summary>
        /// User input message sent to the model. null if no input was sent.
        /// </summary>
        public ChatMessage? Input { get; set; }

        /// <summary>
        /// Message returned.
        /// </summary>
        public ChatMessage? Message { get; set; }

        /// <summary>
        /// Error returned.
        /// </summary>
        public TeamsAIException? Error { get; set; }
    }

    /// <summary>
    /// Status of the prompt response.
    /// </summary>
    public enum PromptResponseStatus
    {
        /// <summary>
        /// Success - The prompt was successfully completed.
        /// </summary>
        Success,
        /// <summary>
        /// Error - An error occurred while completing the prompt.
        /// </summary>
        Error,
        /// <summary>
        /// RateLimited - The request was rate limited.
        /// </summary>
        RateLimited,
        /// <summary>
        /// InvalidResponse - The response was invalid.
        /// </summary>
        InvalidResponse,
        /// <summary>
        /// TooLong - The rendered prompt exceeded the `max_input_tokens` limit.
        /// </summary>
        TooLong
    }
}

File: js/packages/teams-ai/src/types/PromptResponse.ts
import { Message } from '../prompts';

/**
 * Status of the prompt response.
 * @remarks
 * `success` - The prompt was successfully completed.
 * `error` - An error occurred while completing the prompt.
 * `rate_limited` - The request was rate limited.
 * `invalid_response` - The response was invalid.
 * `too_long` - The rendered prompt exceeded the `max_input_tokens` limit.
 */

export type PromptResponseStatus =
    | 'success'
    | 'error'
    | 'rate_limited'
    | 'invalid_response'
    | 'too_long'
    | 'tools_error';

/**
 * Response returned by a `PromptCompletionClient`.
 * @template TContent Optional. Type of the content in the message. Defaults to `unknown`.
 */

export interface PromptResponse<TContent = unknown> {
    /**
     * Status of the prompt response.
     */
    status: PromptResponseStatus;

    /**
     * User input message sent to the model. `undefined` if no input was sent.
     */
    input?: Message<any>;

    /**
     * Message returned.
     * @remarks
     * This will be a `Message<TContent>` object if the status is `success`, otherwise it will be a `string`.
     */
    message?: Message<TContent>;

    /**
     * Error returned.
     */
    error?: Error;
}
